src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1154,6 +1161,34 @@ public WorkManager getWorkManager()
     }
 
     /**
+     * Set whether a raid will happen tonight
+     */
+    public void setRaidWillHappen(final boolean willRaid)
+    {
+        raidWillHappen = willRaid;
+    }
+
+    @Override
+    public boolean getWillRaid()
+    {
+        return raidWillHappen;
+    }
+
+    /**
+     * Set whether a raid has happened.
+     */
+    public void setHasRaided(final boolean hasRaided)
+    {
+        hasRaidHappened = hasRaided;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2696)